src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/PemEncoding.cs

```diff
@@ -387,60 +468,160 @@ public static bool TryWrite(ReadOnlySpan<char> label, ReadOnlySpan<byte> data, S
+        /// </param>
+        /// <returns>
+        /// <see langword="true"/> if <paramref name="destination"/> is large enough to contain
+        /// the PEM-encoded text, otherwise <see langword="false"/>.
+        /// </returns>
+        /// <remarks>
+        /// This method always wraps the base-64 encoded text to 64 characters, per the
+        /// recommended wrapping of IETF RFC 7468. Unix-style line endings are used for line breaks.
+        /// </remarks>
+        /// <exception cref="ArgumentOutOfRangeException">
+        ///   <paramref name="utf8Label"/> exceeds the maximum possible label length.
+        ///   <para>
+        ///       -or-
+        ///   </para>
+        ///   <paramref name="data"/> exceeds the maximum possible encoded data length.
+        /// </exception>
+        /// <exception cref="ArgumentException">
+        /// <para>
+        ///   The resulting PEM-encoded text is larger than <see cref="int.MaxValue"/>.
+        /// </para>
+        /// <para>- or -</para>
+        /// <para>
+        ///   <paramref name="utf8Label"/> contains invalid characters or is malformed UTF-8.
+        /// </para>
+        /// </exception>
+        public static bool TryWriteUtf8(
+            ReadOnlySpan<byte> utf8Label,
+            ReadOnlySpan<byte> data,
+            Span<byte> destination,
+            out int bytesWritten)
         {
-            static int Write(ReadOnlySpan<char> str, Span<char> dest, int offset)
+            if (!Utf8.IsValid(utf8Label) || !IsValidLabel(utf8Label))
+                throw new ArgumentException(SR.Argument_PemEncoding_InvalidLabel, nameof(utf8Label));
+
+            int encodedSize = GetEncodedSize(utf8Label.Length, data.Length);
+
+            if (destination.Length < encodedSize)
             {
-                str.CopyTo(dest.Slice(offset));
-                return str.Length;
+                bytesWritten = 0;
+                return false;
             }
 
-            static int WriteBase64(ReadOnlySpan<byte> bytes, Span<char> dest, int offset)
-            {
-                bool success = Convert.TryToBase64Chars(bytes, dest.Slice(offset), out int base64Written);
+            bytesWritten = WriteCore<byte, Utf8PemEncoder>(utf8Label, data, destination);
+            Debug.Assert(encodedSize == bytesWritten);
+            return true;
+        }
 
-                if (!success)
-                {
-                    Debug.Fail("Convert.TryToBase64Chars failed with a pre-sized buffer");
-                    throw new ArgumentException(null, nameof(destination));
-                }
+        /// <summary>
+        /// Creates an encoded PEM with the given label and data.
+        /// </summary>
+        /// <param name="utf8Label">
+        /// The label to encode.
+        /// </param>
+        /// <param name="data">
+        /// The data to encode.
+        /// </param>
+        /// <returns>
+        /// An array of UTF-8 characters of the encoded PEM.
```

Comment:
Do we say "UTF-8 characters" in other documentation?  I'd probably hedge as "An array containing the bytes representing the UTF-8 encoding of the PEM", since it's UTF-8 Code Units, and "character" is less well defined.